src/services/codefixes/importFixes.ts

```diff
@@ -726,29 +728,16 @@ namespace ts.codefix {
     function getImportCodeActions(context: CodeFixContext): ImportCodeAction[] {
         const importFixContext = convertToImportCodeFixContext(context);
         return context.errorCode === Diagnostics._0_refers_to_a_UMD_global_but_the_current_file_is_a_module_Consider_adding_an_import_instead.code
-            ? getActionsForUMDImport(importFixContext)
+            ? getActionsForUMDImport(importFixContext, importFixContext.isJsxNamespace)
             : getActionsForNonUMDImport(importFixContext, context.program.getSourceFiles(), context.cancellationToken);
     }
 
-    function getActionsForUMDImport(context: ImportCodeFixContext): ImportCodeAction[] {
-        const { checker, symbolToken, compilerOptions } = context;
-        const umdSymbol = checker.getSymbolAtLocation(symbolToken);
-        let symbol: ts.Symbol;
-        let symbolName: string;
-        if (umdSymbol.flags & ts.SymbolFlags.Alias) {
-            symbol = checker.getAliasedSymbol(umdSymbol);
-            symbolName = context.symbolName;
-        }
-        else if (isJsxOpeningLikeElement(symbolToken.parent) && symbolToken.parent.tagName === symbolToken) {
-            // The error wasn't for the symbolAtLocation, it was for the JSX tag itself, which needs access to e.g. `React`.
-            symbol = checker.getAliasedSymbol(checker.resolveName(checker.getJsxNamespace(), symbolToken.parent.tagName, SymbolFlags.Value, /*excludeGlobals*/ false));
-            symbolName = symbol.name;
-        }
-        else {
-            throw Debug.fail("Either the symbol or the JSX namespace should be a UMD global if we got here");
-        }
-
-        return getCodeActionForImport(symbol, { ...context, symbolName, kind: getUmdImportKind(compilerOptions) });
+    function getActionsForUMDImport(context: ImportCodeFixContext, isJsxNamespace: boolean): ImportCodeAction[] {

```

Comment:
I'm still getting used to the way we use intersection types in TS, so pardon me if this is a dumb question: why return the extra property using an intersection type and then pass it on as a separate parameter?  Also, why not actually add it to the type?  To avoid modifying the exported interface?

Reply:
Probably shouldn't have used an intersection type. `convertToImportCodeFixContext` returns 2 values, the `ImportCodeFixContext` and `isJsxNamespace`. Will update.